qconv+relu and qlinear+relu modules

dskhudia opened the pull request:
Summary: adding qconv+relu and qlinear+relu modules in nn/_intrinsic/quantized

Differential Revision: D16505333

jerryzh168 commented on torch/nn/_intrinsic/quantized/modules/conv_relu.py:
```diff
@@ -0,0 +1,33 @@
+        Same as torch.nn.quantized.Conv2d
+
+    """
+    __FLOAT_MODULE__ = NNConvReLU2d
```
it's not there yet for quantized.conv, but I can add this.
also please change to `__FLOAT_MODULE` since that means class attribute

dskhudia replied:
Renamed to __FLOAT_MODULE

jerryzh168 commented on torch/nn/_intrinsic/quantized/modules/conv_relu.py:
```diff
@@ -0,0 +1,33 @@
+    r"""
+    A ConvReLU2d module is a fused module of Conv2d and ReLU
+
+    We adopt the same interface as :class:`torch.nn.quantized.Conv2d`.
```
`torch.nn.Conv2d`?

raghuramank100 replied:
This should be the same as nn.quantized.conv2d.

raghuramank100 reviewed (approved):
Approved, please resolve comments prior to submitting.

dskhudia committed:
qconv+relu and qlinear+relu modules (#23410)

Summary:
adding qconv+relu and qlinear+relu modules in nn/_intrinsic/quantized
Pull Request resolved: https://github.com/pytorch/pytorch/pull/23410

Test Plan:
Extended tests to test these new modules as well

buck test mode/dev caffe2/test:quantized -- 'test_linear_api'  --print-passing-details
```
Running 1 tests
Started new test run: https://our.intern.facebook.com/intern/testinfra/testrun/2251799820197379
      ✓ caffe2/test:quantized - test_linear_api (test_nn_quantized.ModuleAPITest) 4.055 1/1 (passed)
Test output:
> test_linear_api (test_nn_quantized.ModuleAPITest)
> test API functionality for nn.quantized.linear and nn._intrinsic.quantized.linear_relu ... ok
>
> ----------------------------------------------------------------------
> Ran 1 test in 4.056s
>
> OK
Finished test run: https://our.intern.facebook.com/intern/testinfra/testrun/2251799820197379
Summary (total time 10.66s):
  PASS: 1
  FAIL: 0
  SKIP: 0
  FATAL: 0
  TIMEOUT: 0
  OMIT: 0
```

buck test mode/dev caffe2/test:quantized -- 'test_conv_api'  --print-passing-details
```
Running 2 tests
Started new test run: https://our.intern.facebook.com/intern/testinfra/testrun/4785074607089664
      ✓ caffe2/test:quantized - test_conv_api (test_quantized_conv.QuantizedConvTest) 5.195 1/2 (passed)
Test output:
> test_conv_api (test_quantized_conv.QuantizedConvTest)
> Tests the correctness of the conv functional. ... ok
>
> ----------------------------------------------------------------------
> Ran 1 test in 5.195s
>
> OK
      ✓ caffe2/test:quantized - test_conv_api (test_nn_quantized.ModuleAPITest) 10.616 2/2 (passed)
Test output:
> test_conv_api (test_nn_quantized.ModuleAPITest)
> Tests the correctness of the conv module. ... ok
>
> ----------------------------------------------------------------------
> Ran 1 test in 10.616s
>
> OK
Finished test run: https://our.intern.facebook.com/intern/testinfra/testrun/4785074607089664
Summary (total time 17.31s):
  PASS: 2
  FAIL: 0
  SKIP: 0
  FATAL: 0
  TIMEOUT: 0
  OMIT: 0
``

Differential Revision: D16505333

Pulled By: dskhudia

fbshipit-source-id: 7327d10526d50a7056165d52b6942ca50e7624b0
